crates/sui-indexer-alt/src/lib.rs

```diff
@@ -194,6 +205,27 @@ impl Indexer {
         Ok(())
     }
 
+    /// Checks that the first checkpoint override is consistent with the watermark for the pipeline.
+    /// If the watermark does not exist, the override can be anything. If the watermark exists, the
+    /// override must not leave any gap in the data: it can be in the past, or at the tip of the
+    /// network, but not in the future.
+    fn check_first_checkpoint_consistency<P: Processor>(
+        &self,
+        watermark: &Option<CommitterWatermark>,
+    ) -> Result<()> {
+        if let (Some(watermark), Some(first_checkpoint)) = (watermark, self.first_checkpoint) {
+            ensure!(
+                first_checkpoint as i64 <= watermark.checkpoint_hi_inclusive + 1,
+                "For pipeline {}, first checkpoint override {} is too far ahead than watermark {}. This could create gaps in the data.",
```

Comment:
```suggestion
                "For pipeline {}, first checkpoint override {} is too far ahead of watermark {}. This could create gaps in the data.",
```

Reply:
Added TODO. Will follow up in a separate PR.